### PR TITLE
#159236199 Add fitbit integration

### PR DESCRIPTION
**Pull request by RrNn**
#### What does this PR do?
- This PR allows the wger user to pull his/her Fitbit weight data into wger so they view it there.
#### How should this functionality be manually tested?
- Clone the application.
- Run `pip install -r requirements.txt`
- Switch to the branch `ft-add-fitbit-159236199`
- Run the application and go to the `http://127.0.0.1:8000/en/weight/overview/<yourUsername>` endpoint.
- You should see an added button `Sync Fitbit` to the right, on the other buttons, as shown in the images below.
- Click it and you will go to a page that prompts you to connect with Fitbit, then will redirect you to Fitbit if you accept.
- When you reach Fitbit, log in if you are not already logged in, and if you are, select all and click accept.
- Your Fitbit weight data will be pulled into wger and you should see it at `http://127.0.0.1:8000/en/weight/overview/<yourUsername`
- Also visit `http://127.0.0.1:8000/en/exercise/overview/` for the exercises.
- A new Fitbit integration button has been added to the page, click it as shown in one the images below, and follow prompts on the page the follows.
- There you will be able to see your Fitbit Exercises
#### Note: The following environment variables have to be set before testing
- FITBIT_WEIGHT_CLIENT_ID
- FITBIT_WEIGHT_CLIENT_SECRET
- FITBIT_WEIGHT_REDIRECT
- FITBIT_EXERCISE_CLIENT_ID
- FITBIT_EXERCISE_CLIENT_SECRET
- FITBIT_EXERCISE_REDIRECT

#### Relevant PT story
[#159236199](https://www.pivotaltracker.com/story/show/159236199)

screen shot 2018-08-22 at 17 15 02
screen shot 2018-08-22 at 17 14 42

fitbit_task
>`.

### CODE REVIEW

File: wger/core/templates/user/fitbit.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ {% block title %}
+ <div class="text-center">
+ {% trans "Click the button below to sync your fitbit data" %}
+ </div>
+ {% endblock %}
+ 
+ {% block content %}
+
+<a href="{{ fitbit_authentication }}" class="btn btn-primary btn-block"> Connect with Fitbit. </a>
+{% endblock %}
```
Comment (wasswa-derick) [inline wger/core/templates/user/fitbit.html]: Add a newline on line 12.

File: wger/core/urls.py [hunk 2]
```diff
@@ -93,12 +93,15 @@
...
         name='overview'),
+    url(r'^fitbit$', user.fitbit_data,
+        name='fitbit'),
     url(r'',
         user.UserListView.as_view(),
         name='list'),
 
 
 
+
```
Comment (wasswa-derick) [inline wger/core/urls.py]: Erase this line.

File: wger/exercises/templates/exercise/fitbit.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            <td>
+                {% if datum.calories %}
+                {{ datum.calories }}
+                {% else %}
+                {% trans 'N/A' %}
+                {% endif %}
+            </td>
+            
+            <td>
+                {% if datum.description %}
```
Comment (wasswa-derick) [inline wger/exercises/templates/exercise/fitbit.html]: Change `description` to `duration`.

File: wger/exercises/templates/exercise/fitbit.html [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            </td>
+        </tr>
+    {% endfor %}
+</table>
+{% endif %}
+<!-- Since we are not persisting fitbit data,
+reloading will lead to losing it so we prompt the user -->
+{% if data %}
+<script>
+window.onbeforeunload = function (evt) {
```
Comment (wasswa-derick) [inline wger/exercises/templates/exercise/fitbit.html]: Try using jQuery instead of vanilla JS.

File: requirements.txt [hunk 5]
```diff
@@ -34,3 +34,8 @@ django-cors-headers
 
 # Python3 compatibility
 six
+
+#fitbit
+django-fitbit==0.3.0
```
Comment (wasswa-derick) [inline requirements.txt]: Remove `django-fitbit` since it is not used.

Comment (RrNn) [reply requirements.txt]: Done

File: wger/exercises/templates/exercise/fitbit.html (hunk 4 above)
Comment (RrNn) [reply wger/exercises/templates/exercise/fitbit.html]: Done

File: wger/exercises/templates/exercise/fitbit.html (hunk 3 above)
Comment (RrNn) [reply wger/exercises/templates/exercise/fitbit.html]: Done

File: wger/core/urls.py (hunk 2 above)
Comment (RrNn) [reply wger/core/urls.py]: Done

File: wger/core/templates/user/fitbit.html (hunk 1 above)
Comment (RrNn) [reply wger/core/templates/user/fitbit.html]: Done

**Pull request merged**